functions/api/transactions.js

```diff
@@ -0,0 +1,87 @@
+const constants = require('../common/constants');
+const logger = require('../middleware/logger');
+const admin = require('firebase-admin');
+const express = require('express');
+const router = express.Router();
+const cors = require('cors');
+const url = require('url');
+const db = admin.firestore();
+
+/**
+ * @description Route to retireve all transactions under all branches
+ * @returns Json object containing all transactions under all branches
+ */
+router.get("/", async (request, response, next) => {
+    console.log("Retrieving all transactions under all branches");
+    let branchCollectionRef = db.collection('branch');
+    let documents = []
+    documents = await branchCollectionRef.listDocuments();
+    const{user,product,fromDate,toDate} = request.query;    
+    let alltransactions = []
+    for(const doc of documents) {
+        var branchtransactions = {}
+        console.log(`fetching document of branch ${doc.id}`);
+        let subCollectionDocRef = branchCollectionRef.doc(doc.id).collection('transactions');
+        if(user){
+            subCollectionDocRef = subCollectionDocRef.where("user","==",user)
+        }
+        if(product){
+            subCollectionDocRef = subCollectionDocRef.where("product","==",product);
+        }
+        if(fromDate){
+            subCollectionDocRef = subCollectionDocRef.where("date",">=",new Date(fromDate));
+        }
+        if(toDate){
+            subCollectionDocRef = subCollectionDocRef.where("date","<=",new Date(toDate));
+        }
+        let snapshot = await subCollectionDocRef.get()
```

Comment:
looks neat. Some pointers to think:
1. Are we supporting multiple queries, like say combination of user and product. If yes, you would need to change your where clause to support && conditions.
2. If not others, specifically for date I think you need to handle a case when both fromDate and toDate are present.

Reply:
Also, you could add a check to verify the branch exists

Reply:
@neerk28 , I checked for such in firestore references, but what i saw is like when u need to add where it will be one condition each , so here in the if loops am forming the query and it will work in all and combinations.
like user = x && product = y => where("user",'==',x).where("product","==",y).

https://stackoverflow.com/questions/51814343/firestore-query-with-multiple-where-clauses-based-on-parameters

Reply:
@shrikantdange1208 hmm, let me think about that, if branch doesnot exist we get transactions array as empty . Do u feel we should return 404?